service/models.py

```diff
@@ -107,9 +108,41 @@ def find(cls, supplier_id):
         cls.logger.info('Getting supplier with id: {}'.format(supplier_id))
 
         try:
-            return cls.objects(id=supplier_id).first()
-        except DoesNotExist:
+            res = cls.objects(id=supplier_id).first()
+        except ValidationError:
             return None
```

Comment:
```
except DoesNotExist as error:
            raise DataValidationError('Invalid supplier_id: failed to '
                'find a supplier with given supplier_id')
```

Reply:
**ValidationError** here is used to enforce the input **supplier_id** is conform with MongoDB's **_id** format: 
> it must be a 12-byte input or a 24-character hex string
>
Otherwise it would throw a **mongoengine.errors.ValidationError** and crash the server like I did:
![image](https://user-images.githubusercontent.com/17645845/66884815-5ddbc300-efa0-11e9-910d-74db2ba72447.png)
So I assume **ValidationError** should be preserved...

Reply:
Sorry my fault, I copy from code and forgot to change it to ValidationError.

Reply:
No worries! :D